core/src/expressions/expression.cc

```
@@ -29,103 +30,208 @@
+  int cell_size = array_schema_->cell_size(first_attribute_id);
+  assert(type_size == cell_size);
+
+  int number_of_cells = buffer_sizes[0]/cell_size;
+  
+  std::vector<int> cells_to_be_dropped;
+
+  try {
+    print_parser_varmap(parser_);
+    parser_->SetExpr(expression_);
+    print_parser_varmap(parser_);
+  } catch (mup::ParserError const &e) {
+    EXPRESSION_ERROR("Parser SetExpr error: " + e.GetMsg());
+    return TILEDB_EXPR_ERR;
+  }
+  
+  for (int i_cell = 0; i_cell < number_of_cells; i_cell++) {
+    for (int i = 0; i < attributes_.size(); ++i) {
+      if (attribute_map_.find(attributes_[i]) != attribute_map_.end()) {
+        int attribute_id = array_schema_->attribute_id(attributes_[i]);
+        int attribute_type = array_schema_->type(attribute_id);
+        switch (attribute_type) {
+          case TILEDB_CHAR: {
+            char val = *(reinterpret_cast<char *>(buffers[i]) + i_cell);
+            attribute_map_[attributes_[i]] = val;
+            break;
+          }
+          case TILEDB_INT32: {
+            int val = *(reinterpret_cast<int *>(buffers[i]) + i_cell);
+            attribute_map_[attributes_[i]] = val;
+            break;
+          }
+          case TILEDB_FLOAT32: {
+            float val = *(reinterpret_cast<float *>(buffers[i]) + i_cell);
+            attribute_map_[attributes_[i]] = val;
+            break;
+          }
+        }
+      }
+    }
+    
+    try {
+      mup::Value ret = parser_->Eval();
+#ifdef DEBUG
+      // print the result
+      mup::console() << "Ret=" << ret << std::endl;
+#endif
+      // drop the cell from buffer if the return type is a boolean
+      if (drop_cell(ret)) {
+        cells_to_be_dropped.push_back(i_cell);
+      }
+    } catch (mup::ParserError const &e) {
+      EXPRESSION_ERROR("Parser evaluate error, possibly due to bad filter expression: " + "\n\t" + e.GetMsg());
+      return TILEDB_EXPR_ERR;
     }
   }
 
+  // TODO: support only for sparse arrays intially.
```

Should we warn (or do we already elsewhere) if someone tries this with dense arrays?

---

We are not warning yet, it is silently ignored now. Will add a message here. Thanks.